### PR TITLE
updated to utf8proc v2.0.1, gives support for Unicode 9

### PR DESCRIPTION
**Pull request by stevengj**
utf8proc was recently updated to support Unicode 9, and we had to tag a new release of utf8proc anyway because the old version's tests were failing due to its out-of-date Unicode tables (JuliaLang/utf8proc#72).

It seems like a good thing to bump the version in Julia 0.5, so that we get Unicode-9 support (and other improvements, like much smaller Unicode tables).

### CODE REVIEW

**Comment by stevengj** [comment]
cc @Keno

**Comment by Keno** [comment]
LGTM assuming CI passes.

File: deps/utf8proc.version [hunk 1]
```diff
@@ -1,2 +1,2 @@
-UTF8PROC_BRANCH=v1.3
-UTF8PROC_SHA1=85789180158ac7fff85b9f008828d6ac44f072ea
+UTF8PROC_BRANCH=v2.0.1
+UTF8PROC_SHA1=a1fe9955bbc75ffb923c1219bf58befd2688e34c
```
Comment (tkelman) [inline deps/utf8proc.version]: update the checksums

Comment (stevengj) [inline deps/utf8proc.version]: Oh, I always forget about these.  Is there a `make` target to automatically create the new checksum?

File: base/strings/utf8proc.jl [hunk 2]
```diff
@@ -184,6 +184,12 @@ end
 isgraphemebreak(c1::Char, c2::Char) =
     ccall(:utf8proc_grapheme_break, Bool, (UInt32, UInt32), c1, c2)
 
+# Stateful grapheme break required by Unicode-9 rules: the string
+# must be processed in sequence, with state initialized to Ref{Int32}(0).
+# Requires utf8proc v2.0 or later.
+isgraphemebreak(c1::Char, c2::Char, state::Ref{Int32}) =
+    ccall(:utf8proc_grapheme_break_stateful, Bool, (UInt32, UInt32, Ref{Int32}), c1, c2, state)
```
Comment (tkelman) [inline base/strings/utf8proc.jl]: update the readme if this means the minimum utf8proc version is now 2.0

File: deps/utf8proc.version (hunk 1 above)
Comment (tkelman) [inline deps/utf8proc.version]: it should happen automatically in configure-utf8proc or one of those, as long as you aren't downloading it in git mode (which is not the default any more)

Comment (stevengj) [inline deps/utf8proc.version]: Okay, got it.

File: base/strings/utf8proc.jl (hunk 2 above)
Comment (stevengj) [inline base/strings/utf8proc.jl]: done

File: deps/utf8proc.version (hunk 1 above)
Comment (tkelman) [inline deps/utf8proc.version]: and delete the old ones

Comment (stevengj) [inline deps/utf8proc.version]: Done.

**Comment by stevengj** [comment]
Thanks to Unicode 9, we will now be able to properly name our "John Travolta" variables
http://www.unicode.org/emoji/charts/full-emoji-list.html#1f57a_1f3fb
but we still can't use a superscript *q*.

**Comment by tkelman** [comment]
osx failure was a libgit2 issue because i got impatient merging a different pr

File: base/strings/utf8proc.jl [hunk 3]
```diff
@@ -194,28 +200,30 @@ eltype{S}(::Type{GraphemeIterator{S}}) = SubString{S}
...
-        n += isgraphemebreak(c0, c)
+        n += isgraphemebreak(c0, c, state)
         c0 = c
     end
     return n
 end
 
-start(g::GraphemeIterator) = start(g.s)
-done(g::GraphemeIterator, i) = done(g.s, i)
+start(g::GraphemeIterator) = (start(g.s), Ref{Int32}(0))
```
Comment (tkelman) [inline base/strings/utf8proc.jl]: looks like this change broke StringDistances.jl, cc @matthieugomez http://pkg.julialang.org/logs/StringDistances_0.5.log

Comment (stevengj) [inline base/strings/utf8proc.jl]: Probably a good idea not to rely on the contents of iterators.  (Even for strings, the iterator may cease to be an integer someday: #9297.)

Comment (matthieugomez) [inline base/strings/utf8proc.jl]: Thanks for the heads up. I'll update the package once the release candidate for Julia 0.5 is out.

**Pull request merged**